core/genesis.go

```diff
@@ -184,10 +184,12 @@ func SetupGenesisBlock(db ethdb.Database, genesis *Genesis) (*params.ChainConfig
 	if genesis.Config == nil {
 		return nil, errGenesisNoConfig
 	}
-	// Make sure genesis gas limit is consistent
-	gasLimitConfig := genesis.Config.FeeConfig.GasLimit.Uint64()
-	if gasLimitConfig != genesis.GasLimit {
-		return nil, fmt.Errorf("gas limit in fee config (%d) does not match gas limit in header (%d)", gasLimitConfig, genesis.GasLimit)
+	// Make sure genesis gas limit is consistent in SubnetEVM fork
+	if genesis.Config.IsSubnetEVM(new(big.Int).SetUint64(genesis.Timestamp)) {
```

Comment:
This should always happen. Not just after SubnetEVM fork IMO

Reply:
It breaks a few tests, so going to leave as is for now.

Reply:
Put up a fix, but this check is required if we want to support pre-SubnetEVM upgrade functionality.

Reply:
Need to revert the last two commits if we want to maintain this.